starter-code/src/App.js

```diff
@@ -1,18 +1,78 @@
-import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { Component } from "react";
+import "bulma/css/bulma.css";
+import "./App.css";
+import foods from "./foods.json";
+import FoodBox from "./components/FoodBox";
+import FormAddFoodBox from "./components/FormAddFoodBox";
+import SearchBar from "./components/SearchBar";
+import TodaysFoods from "./components/TodaysFoods";
 
 class App extends Component {
+  state = {
+    isActive: false,
+    foods: foods,
+    todaysFoods: [],
+    totalCal: 0
+  };
+
+  handleClick = e => {
+    this.setState({ isActive: true });
+  };
+
+  addMyFood = v => {
+    const copy = [...this.state.todaysFoods];
+    copy.push(v);
+    const total = copy
+      .map(item => item.calories * item.quantity)
+      .reduce((prev, next) => prev + next);
+    this.setState({ todaysFoods: copy, totalCal: total });
+  };
+
+  addProduct = info => {
+    this.setState({ foods: [...this.state.foods, info], isActive: false });
+  };
+
+  handleChange = input => {
+    const searchValue = input.value;
+    let filtered = foods.filter(f =>
+      f.name.toLowerCase().includes(searchValue.toLowerCase())
+    );
+    this.setState({ foods: filtered });
+  };
+
   render() {
     return (
       <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
+        <div className="container">
+          <h1 className="title">IronNutrition</h1>
+          <SearchBar clbk={this.handleChange} />
+          <button className="addFoodBox" onClick={this.handleClick}>
+            Create FoodBox
+          </button>
+        </div>
+        <div className="form">
+          {this.state.isActive && <FormAddFoodBox clbk={this.addProduct} />}
+        </div>
+        <div className="columns">
+          <div className="column">
+            {this.state.foods.map((f, i) => (
```

Comment:
you should implement the filter just before the map: this.state.food.filter(value of the search).map().
Thus you can keep all the food created.

And if the search is empty, it will display all the food.